### PR TITLE
docs: documentation of compiler, config, explorer

### PR DESCRIPTION
**Pull request by Ninjagod1251**
### What I did

made doc strings for compiler, config, explorer

fixes: #

### How I did it

I could not have made it without 
Chunky Monkey by Ben and Jerry
BBM playlist number 6 by Two Friends 

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

File: src/ape/api/compiler.py [hunk 1]
```diff
@@ -14,7 +14,12 @@ class CompilerAPI:
     @property
     @abstractmethod
     def name(self) -> str:
-        ...
+        """
+        Name converts raw intput into string
```
Comment (unparalleled-js) [inline src/ape/api/compiler.py:18]:
```suggestion
        The name of the compiler.
```

File: src/ape/api/config.py [hunk 2]
```diff
@@ -17,6 +17,9 @@ class ConfigItem:
     """
 
     def serialize(self) -> Dict:
+        """
+        Takes raw data and dictionize then serialize it.
```
Comment (unparalleled-js) [inline src/ape/api/config.py:21]:
```suggestion
        Convert to a dict.
```

File: src/ape/api/config.py [hunk 3]
```diff
@@ -43,5 +46,9 @@ def __getitem__(self, attrname: str) -> Any:
 
 
 class ConfigDict(ConfigItem):
+    """
+    A class for Config Dictionary
```
Comment (unparalleled-js) [inline src/ape/api/config.py:50]:
```suggestion
    A config class that is generic and key-value based.
```

File: src/ape/api/explorers.py [hunk 4]
```diff
@@ -14,8 +14,24 @@ class ExplorerAPI:
 
     @abstractmethod
     def get_address_url(self, address: str) -> str:
-        ...
+        """
+        Get address url given an address.
```
Comment (unparalleled-js) [inline src/ape/api/explorers.py:18]:
```suggestion
        Get the address URL for the given address.
```

File: src/ape/api/explorers.py [hunk 5]
```diff
@@ -14,8 +14,24 @@ class ExplorerAPI:
...
+
+        Args:
+            address (`str`): Value of address
```
Comment (unparalleled-js) [inline src/ape/api/explorers.py:21]:
```suggestion
            address (str): Value of address
```

File: src/ape/api/explorers.py [hunk 6]
```diff
@@ -14,8 +14,24 @@ class ExplorerAPI:
...
+
+        Returns:
+            str
+        """
 
     @abstractmethod
     def get_transaction_url(self, transaction_hash: str) -> str:
-        ...
+        """
+        Get transaction url given the transaction.
```
Comment (unparalleled-js) [inline src/ape/api/explorers.py:30]:
```suggestion
        Get the transaction URL for the given transaction.
```

File: src/ape/api/explorers.py [hunk 7]
```diff
@@ -14,8 +14,24 @@ class ExplorerAPI:
...
+
+        Args:
+            transaction_hash (`str`): Value of transaction hash.
```
Comment (unparalleled-js) [inline src/ape/api/explorers.py:33]:
```suggestion
            transaction_hash (str): The transaction hash.
```

File: src/ape/api/compiler.py [hunk 8]
```diff
@@ -14,7 +14,12 @@ class CompilerAPI:
     @property
     @abstractmethod
     def name(self) -> str:
-        ...
+        """
+        The name of the compiler.
```
Comment (fubuloubu) [inline src/ape/api/compiler.py:18]: Might be good to add how this is used in the codebase, which.... I don't know if it is lol

File: src/ape/api/config.py [hunk 9]
```diff
@@ -17,6 +17,9 @@ class ConfigItem:
     """
 
     def serialize(self) -> Dict:
+        """
+        Convert to a dict.
```
Comment (fubuloubu) [inline src/ape/api/config.py:21]:
```suggestion
        Serialize the config item into a raw dict format for storing on disk.
```

**Review by fubuloubu — APPROVED** [review]

File: src/ape/api/compiler.py (hunk 8 above)
Comment (unparalleled-js) [reply src/ape/api/compiler.py:18]: `name` sounds like something that can be in our base API class (have the pydantic work in mind a little)

Comment (fubuloubu) [reply src/ape/api/compiler.py:18]: Yes but I don't think it does anything right now

I think the target might have been for the `compilers` field of the manfiest

Comment (unparalleled-js) [reply src/ape/api/compiler.py:18]: - [ ] let's remove this doc str

File: src/ape/api/explorers.py [hunk 10]
```diff
@@ -14,8 +14,24 @@ class ExplorerAPI:
 
     @abstractmethod
     def get_address_url(self, address: str) -> str:
-        ...
+        """
+        Get the address URL for the given address.
```
Comment (unparalleled-js) [inline src/ape/api/explorers.py:18]:
```suggestion
        Get an address URL, such as for a transaction.
```

File: src/ape/api/compiler.py (hunk 8 above)
Comment (Ninjagod1251) [reply src/ape/api/compiler.py:18]: So should i just remove the comment?

**Review by fubuloubu — APPROVED** [review]

**Review by fubuloubu — DISMISSED** [review]

**Review by unparalleled-js — APPROVED** [review]

**Review by fubuloubu — APPROVED** [review]

**Pull request merged**